portal/static/js/accountCreation.js

```diff
@@ -0,0 +1,572 @@
+        if (!params) {
+            return false;
+        };
+        var self = this;
+        if (!hasValue(params.apiUrl)) {
+            if (params.callback) {
+                (params.callback).call(self, {"error": "API url is required."});
+            };
+            return false;
+        };
+        var errorMessage = "";
+        self.attempts++;
+        self.params = params;
+        $.ajax ({
+            type: (params.requestType ? params.requestType : "GET"),
+            url: String(params.apiUrl).replace("//", "/"),
+            contentType: "application/json; charset=utf-8",
+            dataType: "json",
+            async: params.sync ? false : true,
+            data: params.requestData
+        }).done(function(data) {
+            self.attempts = 0;
+            if (params.callback) {
+                (params.callback).call(self, {"data": data});
+            };
+        }).fail(function(xhr) {
+            if (self.attempts < self.max_attempts) {
+                setTimeout ( function() { self.__request( self.params ); } , $.ajaxSetup().retryAfter );
+            } else {
+                var displayError = i18next.t("Server error occurred updating data.");
+                $("#error_response_text").html(displayError);
+                errorMessage = "Error processing request: " + params.apiUrl;
+                errorMessage += ";  response status code: " + (parseInt(xhr.status) === 0 ? "request timed out/network error": xhr.status);
+                errorMessage +=";  response text: " + (hasValue(xhr.responseText) ? xhr.responseText : "no response text returned from server");
+                tnthAjax.reportError(this.userId, "/patients/patient-profile-create", errorMessage, true);
+                self.attempts = 0;
+                if (params.callback) {
+                    (params.callback).call(self, {"error": displayError});
+                };
+            };
+        });
+        return errorMessage;
+    };
+    this.__setAccount = function() {
+        var orgIDs = $("#userOrgs input:checked").map(function(){
+            return { organization_id: $(this).val() };
+        }).get();
+
+        var _accountArray = {};
+        _accountArray["organizations"] = orgIDs;
+        if (this.roles) {
+            _accountArray["roles"] =  this.roles;
+        };
+        _accountArray["consents"] = this.getConsents();
+
+        //note, this will call put/demographics to update demographics after
+        this.__request({"apiUrl": "/api/account", "requestType": "POST", "requestData": JSON.stringify(_accountArray), "sync": true, "callback": this.__setDemo});
+    };
+    this.__setDemo = function(returnedData) {
+        var responseData = returnedData && returnedData.data? returnedData.data : null;
+        var self = this;
+        if (responseData) {
+
+            self.userId = responseData["user_id"];
+
+            if (isNaN(self.userId)) {
+                self.__handleError("Invalid user id: " + self.userId);
```

Comment:
Tag for translation please :)
You also might wanna give this a try- my best attempt at interpolation (working around i18next):
```
self.__handleError(i18next.t("Invalid user id: %d").replace("%d", self.userId));
```